org.lflang/src/org/lflang/federated/FedASTUtils.java

```diff
@@ -425,10 +425,26 @@ public static void makeCommunication(
         // these reactions to appear only in the federate whose bank ID matches.
         generator.setReactionBankIndex(r1, leftBankIndex);
         generator.setReactionBankIndex(r2, rightBankIndex);
+        
+        // Get the serialization method
+        var serialization = SupportedSerializations.NATIVE;
+        if (connection.getSerialization() != null) {
+            serialization = SupportedSerializations.valueOf(
+                    connection.getSerialization().getSerialization().toUpperCase()
```

Comment:
This looks a little weird. Maybe adjust the grammar so that it's `connection.getSerializer().getName()` or something along those lines?

Reply:
Yes that sounds better.